clients/src/main/java/org/apache/kafka/common/requests/ProduceResponse.java

```diff
@@ -204,118 +75,78 @@ public ProduceResponse(Map<TopicPartition, PartitionResponse> responses) {
      * @param throttleTimeMs Time in milliseconds the response was throttled
      */
     public ProduceResponse(Map<TopicPartition, PartitionResponse> responses, int throttleTimeMs) {
-        this.responses = responses;
-        this.throttleTimeMs = throttleTimeMs;
+        this(new ProduceResponseData()
+            .setResponses(responses.entrySet()
+                .stream()
+                .collect(Collectors.groupingBy(e -> e.getKey().topic()))
+                .entrySet()
+                .stream()
+                .map(topicData -> new ProduceResponseData.TopicProduceResponse()
```

Comment:
Not required, but this would be easier to follow up if we had some helpers.

Reply:
Pardon me. why it is not required?

Reply:
Oh, I was just emphasizing that it is a matter of taste. It's up to you if you agree or not.